### PR TITLE
feat: add shell.nix and fix deploy script 

### PR DESCRIPTION
**Pull request by vdovhanych**
Fixes the deploy script and add shell.nix with some basic settings.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -6,15 +6,15 @@ stages:
...
   stage: deploy
   needs: []
-  before_script: []
   variables:
     GIT_STRATEGY: clone
   only:
     refs:
       - master
   when: manual
+  before_script: []
```
Comment (tsusanka) [inline .gitlab-ci.yml:15]: Why move this?

File: shell.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{ };
+
+stdenv.mkDerivation {
+  name = "webwallet-data";
+  buildInputs = [
+    bash
+    awscli
+    git-lfs
+    git
+    jq
```
Comment (tsusanka) [inline shell.nix]: Do we actually need jq? :)

Comment (vdovhanych) [reply shell.nix]: Guess we won't need it. I will remove it then.

File: .gitlab-ci.yml (hunk 1 above)
Comment (vdovhanych) [reply .gitlab-ci.yml:15]: just my ocd, i like when before script is just before the script section 🙈

File: shell.nix (hunk 2 above)
Comment (vdovhanych) [reply shell.nix]: done

**Review by tsusanka — APPROVED** [review]

**Comment by vdovhanych** [comment]
Is it ok now ?

**Review by tsusanka — APPROVED** [review]

**Pull request merged**